runtime/doc/change.txt

```diff
@@ -985,10 +985,7 @@ inside of strings can change!  Also see 'softtabstop' option. >
 ["x]yy			Yank [count] lines [into register x] |linewise|.
 
 							*Y*
-["x]Y			yank [count] lines [into register x] (synonym for
-			yy, |linewise|).  If you like "Y" to work from the
-			cursor to the end of line (which is more logical,
-			but not Vi-compatible) use ":map Y y$".
+["x]Y			Yank [count] lines from the cursor to the end of line.
```

Comment:
The doc for `:help Y` would not change, rather the pattern would be something like this: 

1. mention the default mapping next to the (legacy) builtin.
2. add a tag like `*Y-default*`

```
*Y-default*
This command is mapped to "y$" by default. |nvim-default-mappings|
```